### ISSUE TITLE
Use of double-precision values in fragment shader leads to software-rendering fallback on M1 GPUs

### ISSUE THREAD

**Issue by bschwind** [glium/glium#1908]
This is more of an oddity with the Apple M1 GPU than an actual issue with glium, apologies for the issue. I just wanted a place to note this quirk because I couldn't find anything at all in my searches.

If you try to use a `double` value in a fragment shader, let's say in the `teapot` example, you'll get the following output:

```
FALLBACK (log once): Fallback to SW vertex processing because buildPipelineState failed
FALLBACK (log once): Fallback to SW fragment processing because buildPipelineState failed
FALLBACK (log once): Fallback to SW vertex processing, m_disable_code: 1000
FALLBACK (log once): Fallback to SW fragment processing, m_disable_code: 1000000
FALLBACK (log once): Fallback to SW vertex processing in drawCore, m_disable_code: 1000
FALLBACK (log once): Fallback to SW fragment processing in drawCore, m_disable_code: 1000000
```

It'll then continue to run very very slowly, in software mode.

Running the `info` example:

```
glium $ cargo run --release --example info
   Compiling glium v0.29.0 (/Users/brian/projects/glium)
    Finished release [optimized] target(s) in 1.70s
     Running `target/release/examples/info`
OpenGL context version: 4.1 Metal - 71.0.7
OpenGL context flags: forward-compatible
OpenGL profile mask: core
OpenGL robustness strategy: none
OpenGL context renderer: Apple M1
OpenGL context vendor: Apple
```

**Comment by micahnyc** [comment]
Wanted to chime in since this was the only page I found on google around this topic. I also had this issue. 

Since metal doens't support double values in shaders, the shaders we not able to convert to metal, and thus software rendering was being used. By converting my shaders from double to float, (and also updating the c++ code that was feeding the shaders) I was able to get hardware rendering again with fps going from 6 to 190.
